jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -57,6 +69,8 @@
 
    private AbstractJCacheListenerAdapter<K,V> listenerAdapter;
 
+   private ConcurrentMap<List<Object>, Queue<CountDownLatch>> latchesByEventSource = new ConcurrentHashMap<>();
```

Comment:
@isavin Apologies for the delay, getting back to reply on this.

I'd not use a List to represent a tuple in Java. Lists are bad tuples for multiple reasons including: lack of type safety with heterogeneous types as elements, it can increase/decrease in size at runtime, and you can mutate each element. There's nothing to help or protect againts all these things. Arrays's are barely better since you cannot change their size that easily but suffer from the other problems.

The simplest thing, and the most type safe one, in spite of the verbosity, is to create a class to hold those 3 elements. In fact, you could call this class Tuple or something similar and make it somehow generic. You can use Scala's `scala.Tuple3` case class as inspiration.

Also, what do you mean by `cache`? The cache name? Or actual cache object?